exporters/etw/README.md

```diff
@@ -0,0 +1,201 @@
+
+- api/include/
+- exporters/etw/include/
+- sdk/include/
+
+Code that instantiates ETW TracerProvider, subsequently obtaining a Tracer bound to `OpenTelemetry-ETW-Provider`,
+and emitting a span named `MySpan` with attributes on it, as well as `MyEvent` within that span.
+
+```cpp
+
+#include <map>
+#include <string>
+
+#include "opentelemetry/exporters/etw/etw_tracer_exporter.h"
+
+using namespace OPENTELEMETRY_NAMESPACE;
+using namespace opentelemetry::exporter::ETW;
+
+// Supply unique instrumentation name (ETW Provider Name) here:
+std::string providerName = "OpenTelemetry-ETW-Provider";
+
+exporter::ETW::TracerProvider tp;
+
+int main(int argc, const char* argv[])
+{
+  // Obtain a Tracer object for instrumentation name.
+  // Each Tracer is associated with unique TraceId.
+  auto tracer = tp.GetTracer(providerName, "TLD");
+
+  // Properties is a helper class in ETW namespace that is otherwise compatible
+  // with Key-Value Iterable accepted by OpenTelemetry API. Using Properties
+  // should enable more efficient data transfer without unnecessary memcpy.
+
+  // Span attributes
+  Properties attribs =
+  {
+    {"attrib1", 1},
+    {"attrib2", 2}
+  };
+
+  // Start Span with attributes
+  auto span = tracer->StartSpan("MySpan", attribs);
+
+  // Emit an event on Span
+  std::string eventName = "MyEvent1";
+  Properties event =
+  {
+    {"uint32Key", (uint32_t)1234},
+    {"uint64Key", (uint64_t)1234567890},
+    {"strKey", "someValue"}
+  };
+  span->AddEvent(eventName, event);
+
+  // End Span.
+  span->End();
+
+  // Close the Tracer on application stop.
+  tracer->CloseWithMicroseconds(0);
+
+  return 0;
+}
+```
+
+Note that different `Tracer` objects may be bound to different ETW destinations.
+
+## Build options and Compiler Defines
+
+While including OpenTelemetry C++ SDK with ETW exporter, the customers are in complete control of
+what options they would like to enable for their project using `Preprocessor Definitions`.
+
+These options affect how "embedded in application" OpenTelemetry C++ SDK code is compiled:
+
+| Name        | Description                                                      |
+|---------------------|------------------------------------------------------------------------------------------------------------------------|
+| HAVE_CPP_STDLIB   | Use STL classes for API surface. This option requires at least C++17. C++20 is recommended.              |
+| HAVE_GSL      | Use [Microsoft GSL](https://github.com/microsoft/GSL) for `gsl::span` implementation. Library must be in include path. |
+| HAVE_ABSEIL_VARIANT | Use `absl::variant` instead of `std::variant`                                      |
+| HAVE_TLD      | Use ETW/TraceLogging Dynamic protocol. This is the default implementation.                       |
+| HAVE_CSTRING_TYPE   | Allow passing C-string type `const char *` to API calls.                                 |
+
```

Comment:
Nicely summarized the build options. Few of the options ( HAVE_ABSEIL_VARIANT, HAVE_CPP_STDLIB, HAVE_ABSEIL_VARIANT ) are applicable as build instructions for  `opentelemetry-cpp`, do you think we can move them there ( https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md#building-as-standalone-cmake-project ), give it's reference and keep rest of them here?

Reply:
Yes, but I want to have a separate document created that covers ALL options. Maybe a separate section in `INSTALL.md` as you are suggesting. For this one - I tried to keep it scoped to what is relevant in this concrete usage scenario.

Reply:
Besides the description for the options, could we add an extra column to the the pros and even cons of using each option to help users to make decision?

Reply:
I like the idea of elaborating on what this option is doing. But maybe in that same description column.

Reply:
I added a few more sentences. `HAVE_CPP_STDLIB` has its own separate detailed design doc here:
https://github.com/open-telemetry/opentelemetry-cpp/blob/main/docs/building-with-stdlib.md